File: model/src/main/java/com/apeiron/igor/model/memory/action/EnemyPlayerAttack.java
package com.apeiron.igor.model.memory.action;

import com.apeiron.igor.model.db.Person;
import com.apeiron.igor.model.db.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class EnemyPlayerAttack implements Acton {
    public String actionName = "EnemyPlayerAttack";
    public Integer xAbscissa;
    public Integer yOrdinate;
    public Person person;

    public User user;
}

File: web-client/src/main/js/igor-client/src/app/model/game-events/enemy-player-attack.ts
import {Person} from "../person";

export class EnemyPlayerAttack {
  public actionName:String = "EnemyPlayerAttack";
  public xAbscissa:number = null;
  public yOrdinate:number = null;
  public person:Person = null;

  constructor(xAbscissa: number, yOrdinate: number, person: Person) {
    this.xAbscissa = xAbscissa;
    this.yOrdinate = yOrdinate;
    this.person = person;
  }
}
